packages/synapse-react-client/src/components/EntityFinder/EntityFinder.tsx

```diff
@@ -42,10 +42,6 @@ const DEFAULT_SELECTABLE_TYPES = Object.values(EntityType)
 const TABLE_DEFAULT_VISIBLE_TYPES = Object.values(EntityType)
 const TREE_DEFAULT_VISIBLE_TYPES = [EntityType.PROJECT, EntityType.FOLDER]
 
-// In the map used to track selections, we use -1 to denote 'selected without version'
-// This is necessary because undefined is returned by map.get when the item is not in the map
-export const NO_VERSION_NUMBER = -1
```

Comment:
Updated the selected type from a map of (entityID, versionNumber) to a map of (entityID, Reference), so we can remove this hacky convention